src/jit/lowerxarch.cpp

```diff
@@ -307,6 +307,7 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
                         GenTreeLclVarCommon* lclVarPtr = tree->gtOp.gtOp1->AsLclVarCommon();
                         LclVarDsc* varDsc = &(compiler->lvaTable[lclVarPtr->gtLclNum]);
                         assert(varDsc->lvIsMultiRegArgOrRet);
+                        varDsc->lvDoNotEnregister = true;
```

Comment:
Should We be using lvaSetVarDoNotEnregister(lclVarPtr->gtLclNum, DNER_IsStruct)?